core/jobs/transforms/job_result_transforms.py

```
@@ -22,7 +22,93 @@
+    def __init__(
+        self,
+        pass_tag: str,
+        *fail_tags: str,
+        prefix: str = '',
+        label: str | None = None,
+    ) -> None:
+        if (pass_tag := pass_tag or 'main') in fail_tags:
+            raise ValueError(f'{pass_tag=!r} must not be one of {fail_tags=!r}')
+        super().__init__(label=label)
+        self.pass_tag = pass_tag
+        self.fail_tags = tuple(sorted({tag for tag in fail_tags}))
```

Ruff (0.15.17)

[warning] Unnecessary set comprehension (rewrite using `set()`)

Rewrite using `set()`

(C416)